# Oem7 Receiver initialization and configuration Commands, sent to Oem7 as-is.
# Refer to Oem7 Receiver documentation. 
#
# 
# Lines marked as "Do Not Remove" are needed to ensure correct operation.
# Some ROS messages are composed from a combination of Oem7 messages.
# It is critical to log individiual Oem7 message to ensure correct generation and field population.
#
# Message generation/composition as as follows:
# 
# ---------------------------------------------------------------------------------------------------
# 'GPSFix': 
#   BESTPOSB
#   BESTVELB
#   INSPVASB (on IMU-enabled units only, not needed on non-IMU units)
#   PSDROP2B (on IMU-enabled units only, not needed on non-IMU units)
#
#  Output rate: the highest rate of:  
#   BESTPOSB, BESTVELB, INSPVASB. 
# Typically, INSPVASB is logged at the highest rate and thus generates GPSFix
#
# -----------------------------------------------------------------------------------------------------
# NavSatFix: published at the same rate as GPSFix; data derived from:
#    BESTPOSB
#
# It is recommended to use GPSFix instead of NavSatFix.
#
# -----------------------------------------------------------------------------------------------------
# 'Imu':
#   INSPVASB
#   INSSTDEVB
#   CORRIMUSB or IMURATECORRIMUSB
#
#  Output rate: the highest rate of:  
#   CORRIMUSB or IMURATECORRIMUSB

# Commented out Oem7 commands are for example purposes
receiver_init_commands:
#-----------------------------------------------------------------------------------------------------
# Reset logging

- "UNLOGALL THISPORT"
#- "FRESET"
#- "!PAUSE 10"
#- "SETINSPROFILE LAND"
#- "SETINSTRANSLATION ANT 1<x> <y> <z> <std. x> <std. y> <std. z>
#- ... etc.


#-------------------------------------------------------------------------------------------------------
# Standard Logging Requests

#----------------------------
# Status & Config
- "LOG INSCONFIGB ONCE"  # Log once to obtain initial configuration. Do not remove, needed for IMU messages.
- "LOG INSCONFIGB ONTIME 300" # Periodic update.
- "LOG RXSTATUSB ONCHANGED"

# GNSS and INS Solutions

# Do not remove these messages -------------------------------------------------------------------------- 
- "LOG BESTPOSB ONTIME 0.1" 
- "LOG BESTVELB ONTIME 0.1" 
- "LOG BESTUTMB ONTIME 1"
- "LOG HEADING2B ONNEW"
- "LOG PSRDOP2B ONCHANGED"
- "LOG INSPVASB ONTIME 0.02"
- "LOG INSPVAXB ONTIME 1"
- "LOG CORRIMUSB ONTIME 0.01"
- "LOG INSSTDEVB ONTIME 1"
#----------------------------------------------------------------------------------------------------------

- "LOG TIMEB ONTIME 1" # Can be used as a 'heartbeat' message.

# Post-Processing Messages; not directly used to generate ROS messages.
- "LOG RAWIMUSXB ONNEW"
- "LOG INSUPDATESTATUSB ONNEW"
- "LOG RAWEPHEMB ONNEW"
- "LOG GLOEPHEMERISB ONNEW"
- "LOG BDSEPHEMERISB ONNEW"
- "LOG GALFNAVEPHEMERISB ONNEW"
- "LOG GALINAVEPHEMERISB ONNEW"
- "LOG RANGEB ONTIME 1"







